src/common/framework/test_group.ts

```diff
@@ -338,19 +338,13 @@ class RunCaseSpecific<
         rec.info(new Error('subcase: ' + stringifyPublicParams(subParams)));
         try {
           const params = mergeParams(this.params, subParams);
-          await this.runTest(
-            rec,
-            params,
-            true,
-            getExpectedStatus(
-              new TestQuerySingleCase(
-                selfQuery.suite,
-                selfQuery.filePathParts,
-                selfQuery.testPathParts,
-                <CaseParams>params
-              )
-            )
+          const subcaseQuery = new TestQuerySingleCase(
+            selfQuery.suite,
+            selfQuery.filePathParts,
+            selfQuery.testPathParts,
+            <CaseParams>params
           );
+          await this.runTest(rec, params, true, getExpectedStatus(subcaseQuery));
```

Comment:
Thinking about this again. How much simpler would things be if we didn't support subcase expectations? Probably not enough that it's worth undoing the work. But I don't think we necessarily need them. We can generally keep dimensions out of the subcases if they are useful to suppress on.

Reply:
I'm going to keep it for now as it's nice for the test author to not have to care about what is/is not possible with expectations.